### PR TITLE
feat: Create dind and node images

### PR DESCRIPTION
**Pull request by apburnes**
## Changes proposed in this pull request:
- Add's initial dockerfile for dcind using the hardened base from https://github.com/cloud-gov/common-pipelines

## security considerations
Moving to the common container hardening and scaning processes for the platform

### CODE REVIEW

**Comment by svenaas** [comment]
@apburnes Is there worthwhile testing I can perform at this stage?

**Comment by apburnes** [comment]
@svenaas you could build the images locally 🤷 . This is repo is directly tied to https://github.com/cloud-gov/common-pipelines and that's where we build, scan and audit in CI. Assurance has ideas on adding additional functionality to the common pipeline tooling to define and run tests but that's not available now. There's not much more since the workflow is still so new.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 ## Why this project
 
-Systems, apps, and tools developed and tested using Docker Compose can also be run using Docker Compose from within a Concourse CI task.
+Pages relies on Concourse CI and CloudFoundry Apps to deliver and serve the platform's different components. This is a centralized repository to hold all of the commonly used images which are hardened and scanned using the cloud.gov [common-pipelines](https://github.com/cloud-gov/common-pipelines) container hardening CI framework.
+
+## How this works
+
+Create a directory in the root of the repository for the name of the image and then a subdirectory for the images's version. To create a Node v20 image, create the directories `./node/v20` and then add the `Dockerfile` and any other supporting scripts under the image version directory.
+
+All Dockerfiles will take the `ARG base_image` argument which is a hardened Ubuntu Jammy image based the [common-pipelines OCI Pipeline](https://github.com/cloud-gov/common-pipelines/blob/feat-add-pages-pipeline/container/README.md).
```
Comment (svenaas) [inline README.md:14]: This link is broken, as the branch `feat-add-pages-pipeline` has been [merged](https://github.com/cloud-gov/common-pipelines/pull/53). I suspect linking to main will do:
```suggestion
All Dockerfiles will take the `ARG base_image` argument which is a hardened Ubuntu Jammy image based the [common-pipelines OCI Pipeline](https://github.com/cloud-gov/common-pipelines/blob/main/container/README.md).
```

**Review by svenaas — CHANGES_REQUESTED** [review]
That's about what I figured. It might be helpful to add a little more documentation on build steps to this repo because people are going to be coming in with different levels of knowledge. They're going to need to run 
```
docker build --build-arg base_image={hardened-jammy} .
```
or something like that but do they need to build the hardened Jammy base image first, or can they reference it?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+To add a new image to `pages-images` you will also need to create a new ECR repository for the common pipeline image put step to use. Add the name of the repository to the [repositories list](https://github.com/cloud-gov/cg-provision/blob/main/terraform/stacks/ecr/stack.tf) in Terraform and make sure the update has been applied.
+
+### Running Locally
+
+To get up and running quickly, you can build an image locally using the Ubuntu Jammy (v22.04) image as your `base_image`.
+
+```sh
+# Navigate into the directory of the image and version you want to build
+# Use the general Ubuntu Jammy base image for local development
+docker buildx build . -t <image-tag> --build-arg base_image=ubuntu:22.04
```
Comment (svenaas) [inline README.md:27]: Is there any reason to tag these builds?

Comment (apburnes) [reply README.md:27]: To use the image again locally.

**Comment by svenaas** [comment]
Both images build perfectly happily using general Ubuntu Jammy as the base. I don't know whether it would be illuminating to additionally confirm with the images from ECR, or whether the proof of that will happen in CI.

**Comment by apburnes** [comment]
Added info on how CI works to give you an idea.

**Review by svenaas — APPROVED** [review]

**Pull request merged**